lib/krane/kubernetes_resource/deployment.rb

```diff
@@ -111,6 +111,14 @@ def validate_definition(*, **)
       @validation_errors.empty?
     end
 
+    def predeployed?
+      krane_annotation_value("predeployed") == "true"
+    end
+
+    def kind
+      @definition["kind"]
+    end
+
```

Comment:
Should we move these methods to the parent class, `KubernetesResource`, to reduce duplication? Additionally, it’s worth noting that the `kind` method already exists in the [parent class](https://github.com/powerhome/krane/blame/ca2035255387ee9d0152ae62acfa8dc8c139ea56/lib/krane/kubernetes_resource.rb#L77-L79)